### PR TITLE
[Common] code clean

### PR DESCRIPTION
**Pull request by jaeyun-jung**
fix typo and remove duplicated code to compare file extension.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Review by gichan-jang — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 1]
```diff
@@ -2571,27 +2463,155 @@ accl_hw_get_type (void)
...
+
+/**
+ * @brief Get neural network framework name from given model file. This does not guarantee the framework is available on the target device.
+ * @param[in] model_files the prediction model paths
+ * @param[in] num_models the number of model files
+ * @param[in] load_conf flag to load configuration for the priority of framework
+ * @return Possible framework name (NULL if it fails to detect automatically). Caller should free returned value using g_free().
+ */
+char *
+nnstreamer_filter_detect_framework (const char **model_files,
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c:2515]: const char ** ? char * const * ?

File: gst/nnstreamer/include/nnstreamer_plugin_api_filter.h [hunk 2]
```diff
@@ -504,6 +504,25 @@ nnstreamer_filter_exit (const char *name);
...
+
+/**
+ * @brief Check if the given hw is supported by the framework.
+ * @param[in] name The name of filter sub-plugin.
+ * @param[in] hw Backend accelerator hardware.
+ * @return TRUE if given hw is available.
+ */
+extern int
+nnstreamer_filter_check_hw_availability (const char *name, const accl_hw hw);
+
```
Comment (myungjoo) [inline gst/nnstreamer/include/nnstreamer_plugin_api_filter.h:525]: This is "internal API" for filter subplugins, not API implementations.
You may need to add one more header file, exported for devel, at /gst/nnstreamer/include
OR
you can keep it in the original position; you are migrating API headers only.

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c]: changed to `const char *const*`

**Comment by jaeyun-jung** [comment]
2021-01-20 PR changed - code clean / reduce duplicated code

File: gst/nnstreamer/include/nnstreamer_plugin_api_filter.h (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/include/nnstreamer_plugin_api_filter.h]: I will reconsider to migrate API (headers only. or all implementation to API repo)

**Review by dongju-chae — APPROVED** [review]
LGTM

**Pull request merged**